### ISSUE TITLE
Kafka consumer fails to consume claim-check message in Azure blob

### ISSUE THREAD

**Issue by fubinzh** [pingcap/tiflow#9794]
### What did you do?

1. create kafka changefeed with claim-check on and claim-check message configured to send to Azure blob.
```
[sink.kafka-config.large-message-handle]
large-message-handle-option = "claim-check"
claim-check-storage-uri = "azure://test/claim-check?account-name=qaworkloaddataset&account-key=xxx"
large-message-handle-compression = "lz4"
```
2. Run workload with large message
3. Wait checkpoint advance
4. Run  kafka consumer to consume the message, kafka configuration file is same as that of CDC.

### What did you expect to see?

Kakfa consumer should consume all the message correctly including those sent to azure blob.

### What did you see instead?

CDC changefeed checkpoint advance correctly, and metrics indicates that claim check message sent to azure blob successfully, but consumer fails to consume kafka message.
![6cfa612c-bc90-4ff1-b3b5-6ededa062c41](https://github.com/pingcap/tiflow/assets/7403864/75053132-3b8b-40e4-adcc-231662ef3513)

```
/ # tail -n2 cdc_kafka_consumer.log
[2023/09/22 03:18:23.977 +00:00] [INFO] [mysql_ddl_sink.go:213] ["Exec DDL succeeded"] [sql="CREATE TABLE `large_row_3` (`id` BIGINT PRIMARY KEY,`col_0` VARCHAR(16383),`col_1` VARCHAR(16383),`col_2` VARCHAR(16383),`col_3` VARCHAR(16383),`col_4` VARCHAR(16383),`col_5` VARCHAR(16383),`col_
6` VARCHAR(16383),`col_7` VARCHAR(16383),`col_8` VARCHAR(16383),`col_9` VARCHAR(16383),`col_10` VARCHAR(16383),`col_11` VARCHAR(16383),`col_12` VARCHAR(16383),`col_13` VARCHAR(16383),`col_14` VARCHAR(16383),`col_15` VARCHAR(16383),`col_16` VARCHAR(16383),`col_17` VARCHAR(16383),`col_18` 
VARCHAR(16383),`col_19` VARCHAR(16383),`col_20` VARCHAR(16383),`col_21` VARCHAR(16383),`col_22` VARCHAR(16383),`col_23` VARCHAR(16383),`col_24` VARCHAR(16383),`col_25` VARCHAR(16383),`col_26` VARCHAR(16383),`col_27` VARCHAR(16383),`col_28` VARCHAR(16383),`col_29` VARCHAR(16383),`col_30` 
VARCHAR(16383),`col_31` VARCHAR(16383),`col_32` VARCHAR(16383),`col_33` VARCHAR(16383),`col_34` VARCHAR(16383),`col_35` VARCHAR(16383),`col_36` VARCHAR(16383),`col_37` VARCHAR(16383),`col_38` VARCHAR(16383),`col_39` VARCHAR(16383),`col_40` VARCHAR(16383),`col_41` VARCHAR(16383),`col_42` 
VARCHAR(16383),`col_43` VARCHAR(16383),`col_44` VARCHAR(16383),`col_45` VARCHAR(16383),`col_46` VARCHAR(16383),`col_47` VARCHAR(16383),`col_48` VARCHAR(16383),`col_49` VARCHAR(16383),`col_50` VARCHAR(16383),`col_51` VARCHAR(16383),`col_52` VARCHAR(16383),`col_53` VARCHAR(16383),`col_54` 
VARCHAR(16383),`col_55` VARCHAR(16383),`col_56` VARCHAR(16383),`col_57` VARCHAR(16383),`col_58` VARCHAR(16383),`col_59` VARCHAR(16383),`col_60` VARCHAR(16383),`col_61` VARCHAR(16383),`col_62` VARCHAR(16383),`col_63` VARCHAR(16383))"] [duration=137.721877ms] [namespace=default] [changefee
d=kafka-consumer]
[2023/09/22 03:18:32.419 +00:00] [PANIC] [main.go:659] ["decode message value failed"] [value="\u0004\"M\u0018dp\ufffd`\u0001\u0000\u0000\ufffdi{\"id\":0,\"database\":\"workload\",\"table\":\"large_row_0\",\"pkNames\":[\"id\"],\"isDdl\":false,\"type\":\"INSERT\",\"es\":1695352698489,\"t\
u0013\u0000\ufffd\u000b706462,\"sql\":\"\",\"sqlType\":\ufffd\u0000{-5},\"my\u0016\u0000\ufffd\u000e\"bigint\"},\"old\":null,\"data\":[9\u0000\ufffd\ufffd\"3595761694220428952\"}],\"_tidb\":{\"commitTs\":444426537792700422,\"claimCheckLocation\":\"azure:/test/claim-check/534f057f-a322-41
bd-9c8e-f63b719658fc.json\"}}\u0000\u0000\u0000\u0000\ufffd\ufffdR\ufffd"] [error="Failed to read azure blob file, file info: bucket(container)='test', key='claim-check/534f057f-a322-41bd-9c8e-f63b719658fc.json': context deadline exceeded"] [errorVerbose="context deadline exceeded\nFaile
d to read azure blob file, file info: bucket(container)='test', key='claim-check/534f057f-a322-41bd-9c8e-f63b719658fc.json'\ngithub.com/pingcap/tidb/br/pkg/storage.(*AzureBlobStorage).ReadFile\n\tgithub.com/pingcap/tidb@v1.1.0-beta.0.20230823131104-05aa17143df8/br/pkg/storage/azblob.go:4
35\ngithub.com/pingcap/tiflow/pkg/sink/codec/canal.(*batchDecoder).assembleClaimCheckRowChangedEvent\n\tgithub.com/pingcap/tiflow/pkg/sink/codec/canal/canal_json_decoder.go:145\ngithub.com/pingcap/tiflow/pkg/sink/codec/canal.(*batchDecoder).NextRowChangedEvent\n\tgithub.com/pingcap/tiflo
w/pkg/sink/codec/canal/canal_json_decoder.go:296\nmain.(*Consumer).ConsumeClaim\n\tgithub.com/pingcap/tiflow/cmd/kafka-consumer/main.go:657\ngithub.com/Shopify/sarama.(*consumerGroupSession).consume\n\tgithub.com/Shopify/sarama@v1.38.1/consumer_group.go:868\ngithub.com/Shopify/sarama.new
ConsumerGroupSession.func2\n\tgithub.com/Shopify/sarama@v1.38.1/consumer_group.go:793\nruntime.goexit\n\truntime/asm_amd64.s:1650"] [stack="main.(*Consumer).ConsumeClaim\n\tgithub.com/pingcap/tiflow/cmd/kafka-consumer/main.go:659\ngithub.com/Shopify/sarama.(*consumerGroupSession).consume
\n\tgithub.com/Shopify/sarama@v1.38.1/consumer_group.go:868\ngithub.com/Shopify/sarama.newConsumerGroupSession.func2\n\tgithub.com/Shopify/sarama@v1.38.1/consumer_group.go:793"]
/ #
```

### Versions of the cluster

[root@upstream-ticdc-0 /]# /cdc version
Release Version: v7.4.0-alpha
Git Commit Hash: f5ecaf97cb224a4e886a6f6502b046a3d1b4e542
Git Branch: heads/refs/tags/v7.4.0-alpha
UTC Build Time: 2023-09-20 11:36:28
Go Version: go version go1.21.1 linux/amd64
Failpoint Build: false

**Comment by 3AceShowHand** [comment]
closed by #9778

**Comment by fubinzh** [comment]
verifyed
/close
